File: typescript/src/bebop_handlers/reset.ts
import GameState from '../classes/gameState';
import { ResetMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';

export default class ResetMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = ResetMsg.decode(message);
        GameState.Instance.setResetCount();
        GameState.Instance.signalReset(message);
    }
}

File: dart/lib/bebop_handlers/reset.dart
import 'package:lean_dart_client/classes/game_state.dart';

import 'bebop_handler.dart';
import 'package:lean_dart_client/schemas.g.dart';

class ResetMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    message = ResetMsg.decode(message);
    GameState.instance.setResetCounter();
    GameState.instance.signalReset(message);
  }
}
